amworkflow/geometry/geometry.py

```diff
@@ -210,3 +290,232 @@ def geometry_spawn(self) -> TopoDS_Shape:
+    else:
+        length = side_len
+        overall_length = (
+            1.5 * line_width + (2 * np.cos(angle) + 2) * side_len * honeycomb_num
+        )
+        overall_width = 3 * line_width + 2 * np.sin(angle) * length
+
+    start_point = np.array([0, line_width * 0.5])
+    offset = np.array([line_width * 0.5 + length * 0.5, 0])
+
+    half_points = np.zeros((int((12 + (honeycomb_num - 1) * 10) / 2), 2))
+    half_points[0] = start_point
+    for i in range(honeycomb_num):
+        start = (
+            start_point + offset + np.array([(2 * np.cos(angle) + 2) * length * i, 0])
+        )
+        honeycomb_unit = create_half_honeycomb(start, length, length, angle)
+        half_points[i * 5 + 1 : i * 5 + 6] = honeycomb_unit
+
+    another_half = np.flipud(np.copy(half_points) * [1, -1])
+    points = np.concatenate((half_points, another_half), axis=0)
+    outer_points = np.array(
+        [
+            [0, -overall_width * 0.5],
+            [overall_length, -overall_width * 0.5],
+            [overall_length, overall_width * 0.5],
+            [0, overall_width * 0.5],
+        ]
+    )
+    points = np.concatenate((points, outer_points), axis=0)
+    if expand_factor != 0:
+        cnt = np.mean(points, axis=0)
+        points[2:-6] = (points[2:-6] - cnt) * (1 + expand_factor) + cnt
+    return points
+
+
+def zigzag_infill(
+    overall_length: float,
+    overall_width: float,
+    line_width: float,
+    zigzag_num: int = 1,
+    angle: float = 1.1468,
+    regular: bool = False,
+    side_len: float = None,
+    expand_factor: float = 0,
+):
+    """
+    Create zigzag geometry.
+
+    Args:
+        overall_length: Length of the zigzag infill.
+        overall_width: Width of the zigzag infill.
+        line_width: Width of the zigzag lines.
+        ...
+
+    Returns:
+        np.ndarray: Array of points defining the zigzag geometry.
+    """
+
+    def calculate_lengths_and_widths(angle_length_pair):
+        x_rad = np.radians(angle_length_pair[0])
+        eq1 = (
+            (
+                (2 * np.cos(x_rad)) * angle_length_pair[1]
+                + line_width * np.sin(x_rad) * 2
+            )
+            * zigzag_num
+            + 2 * line_width
+            - overall_length
+        )
+        eq2 = 3 * line_width + 2 * angle_length_pair[1] * np.sin(x_rad) - overall_width
+        return [eq1, eq2]
+
+    def create_half_zigzag(origin, side_length1, side_length2, angle):
+        points = np.zeros((5, 2))
+        points[0] = origin
+        points[1] = origin + np.array(
+            [side_length1 * np.cos(angle), side_length1 * np.sin(angle)]
+        )
+        points[2] = points[1] + np.array([side_length2, 0])
+        points[3] = points[2] + np.array(
+            [side_length1 * np.cos(angle), -side_length1 * np.sin(angle)]
+        )
+        points[4] = points[3] + np.array([side_length2 * 0.5, 0])
+        return points
+
+    if not regular:
```

Comment:
what exaclty is not regular?

Reply:
So for a honeycomb, what I meant "regular" is that all sides are the same and the angle is of course pi/3.